### PR TITLE
ostree commit --tree=tar: Import xattrs from tarballs

### PR DESCRIPTION
**Pull request by wmanley**
If you specify an `xattr_callback` the xattrs will still be taken from
there for now.

**TODO:**

- [x] Test that `--no-xattrs` is respected

### CODE REVIEW

File: src/libostree/ostree-repo-libarchive.c [hunk 1]
```diff
@@ -458,12 +458,43 @@ aic_get_xattrs (OstreeRepoArchiveImportContext *ctx,
...
+               ctx->entry, &name, &value, &size) == ARCHIVE_OK)
+        {
+          g_variant_builder_add (builder, "(@ay@ay)",
+                                 g_variant_new_bytestring (name),
+                                 g_variant_new_fixed_array (G_VARIANT_TYPE("y"),
+                                                            value, size, 1));
+        }
+
+      xattrs = g_variant_builder_end (builder);
+      g_variant_ref_sink (xattrs);
```
Comment (jlebon) [inline src/libostree/ostree-repo-libarchive.c:485]: Minor: we usually inline these two... ahh OK you're following the pattern from below. SGTM!

File: tests/test-libarchive-import.c [hunk 2]
```diff
@@ -453,6 +456,90 @@ test_libarchive_xattr_callback (gconstpointer data)
...
+
+  if (!import_write_and_ref (td->repo, &opts, a, "baz", NULL, &error))
+    goto out;
+
+  /* check contents */
+  if (!spawn_cmdline ("ostree --repo=repo checkout baz import-checkout", NULL))
+    goto out;
+
+  spawn_cmdline ("ostree --repo=repo ls -R -X baz", NULL);
+  spawn_cmdline ("attr -l import-checkout/anotherfile", NULL);
```
Comment (jlebon) [inline tests/test-libarchive-import.c:489]: Left-over from debugging? Or did you mean to leave this around for future debugging as well?

**Review by jlebon — APPROVED** [review]
Nice and simple! Minor comments, but LGTM as is too.
/approve

File: tests/test-libarchive-import.c (hunk 2 above)
Comment (wmanley) [reply tests/test-libarchive-import.c:489]: Leaving it in was intentional.  I've got some plans to adjust these tests in the future.

**Pull request merged**